tools/npm_package_sass_library.bzl

```diff
@@ -0,0 +1,50 @@
+load("@build_bazel_rules_nodejs//:providers.bzl", "NpmPackageInfo")
+load("@io_bazel_rules_sass//:defs.bzl", "SassInfo")
+
+"""
+  Filters a list of files to only return Sass files (with the `.scss` extension).
+"""
+
+def _filter_sass_files(files):
+    return [f for f in files if f.short_path.endswith(".scss")]
+
+"""
+  Rule that extracts Sass sources and its transitive dependencies from a NPM
+  package. The extracted source files are provided with the `SassInfo` provider
+  so that they can be consumed directly as dependencies of other Sass libraries
+  or Sass binaries.
+
+  This rule is helpful when build targets rely on Sass files provided by an external
+  NPM package. In those cases, one wouldn't want to list out all individual source
+  files of the NPM package, but rather glob all needed Sass files from the NPM package.
+"""
+
+def _npm_package_sass_library(ctx):
+    transitive_sources = []
+
+    # Iterate through all specified dependencies and collect Sass files from build
+    # targets that have the `NpmPackageInfo` provider set. The `yarn_install` rule
+    # automatically sets these providers for individual targets in `@npm//<..>`.
+    for dep in ctx.attr.deps:
+        npm_info = dep[NpmPackageInfo]
+        if npm_info != None:
+            filered_files = _filter_sass_files(npm_info.sources.to_list())
+            transitive_sources.append(depset(filered_files))
+
+    # Convert the collected transitive Sass sources to a depset. This is necessary
+    # for proper deduping of dependencies. Performance-wise it's not efficient that
+    # we need to unwrap the depset for NPM packages, but this is necessary to ensure
+    # incrementality and efficient sandboxing for targets relying on these outputs.
+    outputs = depset(transitive = transitive_sources)
+
+    return [
+        DefaultInfo(files = outputs),
+        SassInfo(transitive_sources = outputs),
+    ]
+
+npm_package_sass_library = rule(
```

Comment:
nit: I don't know whether "package" is the correct term here since we aren't actually packaging anything, we're just exposing it. Maybe something like "list_sass_files" instead?

Reply:
It's not about packaging, but rather about providing a sass library for a NPM package. I had something like `list_sass_files` before, but that is not correct given that we provide the `SassInfo` provider and only are concerned with Sass files from NPM packages.

Maybe it would be more clear with `sass_library_for_npm_package`, though that seems like an uncommon rule name in the Bazel ecosystem.

Reply:
@crisbeto Changed it to `npm_sass_library`. Maybe that is better?